src/bounces/mod.rs

```diff
@@ -74,21 +74,18 @@ impl From<BounceError> for Failure {
     }
 }
 
-pub struct Bounces<'a> {
-    db: Box<&'a Db>,
-    limits: &'a BounceLimits,
+pub struct Bounces {
+    limits: BounceLimits,
 }
 
-impl<'a> Bounces<'a> {
-    pub fn new(settings: &'a Settings, db: Box<&'a Db>) -> Bounces<'a> {
+impl Bounces {
+    pub fn new(settings: &Settings) -> Bounces {
         Bounces {
-            db,
-            limits: &settings.bouncelimits,
+            limits: settings.bouncelimits.clone(),
         }
     }
 
-    pub fn check(&self, address: &str) -> Result<(), BounceError> {
-        let bounces = self.db.get_bounces(address)?;
+    pub fn check(&self, address: &str, bounces: Vec<BounceRecord>) -> Result<(), BounceError> {
```

Comment:
So two things here I guess:

1. Now that it's coming in from the outside, let's call the argument `bounce_records` to be a bit more explicit about it and improve greppability.

2. I find the new API for this type a little bit clunky, making the callers do extra legwork in calling the db method on our behalf then passing in the result. Rather than putting `Db` into managed state, can we make it so that only `Bounces` is in managed state and the `Db` instance is retained as a member of `Bounces`? We'd need to put back a bit of the lifetime stuff to make that work, but I think it would be a friendlier interface.

Reply:
I realise, in writing that, my points contradict each other. If we do `2`, which is preferable I think, then we can ignore all the stuff I said about param names here and elsewhere.

Reply:
> can we make it so that only Bounces is in managed state and the Db instance is retained as a member of Bounces? We'd need to put back a bit of the lifetime stuff to make that work, but I think it would be a friendlier interface.

This fact that managed state can only deal with `'static` stuff it makes it really complex to deal with db inside of Bounces.  You can experience the problems with that by running this branch: https://github.com/brizental/fxa-email-service/blob/brizental/inject-config-not-lazy/src/service_main.rs#L42

After chatting @pjenvey it was advised to us to separate the DB and Bounces

> [17:16:23] <pjenvey>		I think that's the easiest way to solve that one	
[17:18:27] <pjenvey>		Providers is a little more involved, what needs to happen w/ it is since rocket mandates all these managed pieces to be 'static, what we really need is all the Providers to own their values they pull from config	
[17:18:56] <pjenvey>		which means a few clones() and the like here and there, which is nice to avoid but really doesn't matter here because it's done one time at startup	
[17:20:30] <pjenvey>		that means you won't be able to toggle changes in your Settings during runtime.. but your that should probably be the case anyway	
[17:20:40] <pjenvey>		(change them and have them propagate)

Reply:
Missed the earlier part from IRC:

> [17:07:55] <pjenvey>		do any of your handlers make raw db calls? solving bounces would be pretty straightforward if it didn't make the db call here	
[17:08:05] <pjenvey>		say the handler made the db call itself, then called bounces

Reply:
> This fact that managed state can only deal with `'static` stuff it makes it really complex to deal with db inside of Bounces.

I think that can be fixed if `Bounces` is generic, so it has a concrete `Db` implementation instead of `Box<Db>`. Then there's no need for any lifetimes and it should just work, right?